samples/web/Continote/public/scripts/my-notes.js

```diff
@@ -0,0 +1,436 @@
+
+      if (!note || note.title === null || note.content === null) {
+        // Invalid  Note, so do nothing.
+        // This should never happen, but just in case.
+        return reject(errorMessage);
+      }
+
+      if (!notesRef_) {
+        // If this occurs, it is likely because the user is not signed in.
+        // However, this should never happen since we set up and tear down the
+        // UI based on the user's Auth state. But, just in case.
+        return reject(errorMessage);
+      }
+
+      var newNoteRef = notesRef_.push();
+      return newNoteRef.set(note).then(function() {
+        return resolve(newNoteRef.key);
+      }).catch(function(error) {
+        return reject(error);
+      });
+    });
+  };
+
+  /**
+   * Deletes the Note with the given key from the Firebase Realtime Database.
+   *
+   * @function
+   * @param {!string} noteKey - The key of the Note from the Firebase Realtime
+   * Database.
+   * @returns {!Promise} - A Promise which either resolves with nothing,
+   * or rejects with an error message.
+   * @const
+   */
+  var deleteNoteFromDatabase_ = function(noteKey) {
+    return new Promise(function(resolve, reject) {
+      var errorMessage = "Cannot delete Note from database";
+
+      if (!noteKey) {
+        // Invalid key for Note, so do nothing.
+        // This should never happen, but just in case.
+        return reject(errorMessage);
+      }
+
+      if (!notesRef_) {
+        // If this occurs, it is likely because the user is not signed in.
+        // However, this should never happen since we set up and tear down the
+        // UI based on the user's Auth state. But, just in case.
+        return reject(errorMessage);
+      }
+
+      var noteRef = notesRef_.child(noteKey);
+      if (!noteRef) {
+        // There is no Note for the signed in user with the provided key.
+        return reject(errorMessage);
+      }
+
+      // Delete the Note from the database.
+      return noteRef.remove();
+    });
+  };
+
+  /**
+   * Opens the Edit Note page to edit the Note from the Firebase Realtime
+   * Database with the given key.
+   *
+   * @function
+   * @param {!string} noteKey - The key of the Note from the Firebase Realtime
+   * Database.
+   * @const
+   */
+  var goToEditNotePageForNoteWithKey_ = function(noteKey) {
+    if (!noteKey) {
+      // Invalid key for Note, so do nothing.
+      // This should never happen, but just in case.
+      return;
+    }
+
+    location.href = "/edit-note.html?noteKey=" + noteKey;
+  };
+
+  /**
+   * Adds an element for the Note with the given key to the Note list, or
+   * updates the element for the Note in the list if one was already there.
+   *
+   * @function
+   * @param {!string} noteKey - The key of the Note from the Firebase Realtime
+   * Database.
+   * @param {!Note} note - The Note from the Firebase Realtime Database.
+   * @const
+   */
+  var displayNoteInList_ = function(noteKey, note) {
+    if (!noteKey) {
```

Comment:
There are a bunch of these simple assertions that could probably be compressed to save space. It can be a bit daunting to look through such long methods :). I would suggest something like this (and similar in other cases):

```
if (!noteKey || !note) {
  // This should never happen, but just in case.
  return;
}
```